components/ide/starthelper/pkg/metrics/metrics.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package metrics
+
+import (
+	"context"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+)
+
+type Metrics struct {
+	IDE    string
+	pusher *push.Pusher
+
+	ctx       context.Context
+	wg        sync.WaitGroup
+	ctxCancel context.CancelFunc
+
+	phaseHistogram *prometheus.HistogramVec
+}
+
+func NewMetrics(IDE string) *Metrics {
+	addr := os.Getenv("SUPERVISOR_ADDR")
+	if addr == "" {
+		addr = "localhost:22999"
+	}
+	pusher := push.New(addr, "codehelper")
+	phaseHistogram := prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "gitpod_ide_starthelper_phase_duration",
+		Help:    "Duration of different phase during code startup",
+		Buckets: []float64{0.01, 0.1, 0.2, 0.5, 2, 5, 10, 30, 60},
+	}, []string{"phase", "ide"})
+
+	pusher.Collector(phaseHistogram)
```

Comment:
is not it going to be expensive if we push it through supervisor again? We wanted to get rid of push gateway from supervisor and push directly to IDE proxy